src/test/scala/org/kibanaLoadTest/helpers/HttpHelper.scala

```diff
@@ -27,43 +30,92 @@ class HttpHelper(appConfig: KibanaConfiguration) {
+      )
+      sampleDataRequest.addHeader("Connection", "keep-alive")
+      sampleDataRequest.addHeader("kbn-version", appConfig.buildVersion)
 
-    val sampleDataResponse = httpClient.execute(sampleDataRequest)
+      val sampleDataResponse = httpClient.execute(sampleDataRequest)
+      statusCode = sampleDataResponse.getStatusLine.getStatusCode
+      responseBody = EntityUtils.toString(sampleDataResponse.getEntity, "UTF-8")
+    } catch {
+      case _: Throwable => {
+        logger.error("Exception occurred during unloading sample data")
+      }
+    } finally {
+      httpClient.close()
+    }
 
-    if (sampleDataResponse.getStatusLine.getStatusCode != 204) {
-      println("Deleting sample data failed")
+    if (statusCode != 204) {
+      throw new RuntimeException(
+        s"Deleting sample data failed: ${responseBody}"
+      )
     }
-    this
   }
 
-  def addSampleData(data: String): HttpHelper = {
-    val sampleDataRequest = new HttpPost(
-      appConfig.baseUrl + s"/api/sample_data/${data}"
-    )
-    sampleDataRequest.addHeader("Connection", "keep-alive")
-    sampleDataRequest.addHeader("kbn-version", appConfig.buildVersion)
+  def addSampleData(data: String): Unit = {
+    var statusCode = 500
+    var responseBody = ""
+    val httpClient = HttpClientBuilder.create.build
+    try {
+      this.loginIfNeeded(httpClient)
+      val sampleDataRequest = new HttpPost(
+        appConfig.baseUrl + s"/api/sample_data/${data}"
+      )
+      sampleDataRequest.addHeader("Connection", "keep-alive")
+      sampleDataRequest.addHeader("kbn-version", appConfig.buildVersion)
 
-    val sampleDataResponse = httpClient.execute(sampleDataRequest)
+      val sampleDataResponse = httpClient.execute(sampleDataRequest)
+      statusCode = sampleDataResponse.getStatusLine.getStatusCode
+      responseBody = EntityUtils.toString(sampleDataResponse.getEntity, "UTF-8")
+    } catch {
+      case _: Throwable => {
+        logger.error("Exception occurred during loading sample data")
+      }
+    } finally {
+      httpClient.close()
+    }
 
-    if (sampleDataResponse.getStatusLine.getStatusCode != 200) {
-      println("Adding sample data failed")
+    if (statusCode != 200) {
+      throw new RuntimeException(
+        s"Adding sample data failed: ${responseBody}"
+      )
     }
-    this
   }
 
-  def closeConnection(): Unit = {
-    httpClient.close()
+  def getStatus(): String = {
+    var responseBody = ""
+    val httpClient = HttpClientBuilder.create.build
+    try {
+      this.loginIfNeeded(httpClient)
+      val statusRequest = new HttpGet(
+        appConfig.baseUrl + "/api/status"
+      )
+      val statusResponse = httpClient.execute(statusRequest)
+      responseBody = EntityUtils.toString(statusResponse.getEntity, "UTF-8")
+    } catch {
+      case _: Throwable => {
+        logger.error("Exception occurred during getting Kibana status")
+      }
+    } finally {
+      httpClient.close()
+    }
+
+    responseBody
```

Comment:
Last line return?  That's what it seems like.

Reply:
exactly. In scala there is no need to use `return` statement. Last line result is returned unless function output is defined as "Unit"